src/Microsoft.TemplateEngine.Cli/TemplateSearch/CliTemplateSearchCoordinator.cs

```diff
@@ -224,18 +230,51 @@ internal string PrintableTotalDownloads
                     }
                     else if (TotalDownloads < 1000)
                     {
-                        return "<1k";
+                        return MinimumDownloadCount;
                     }
                     else
                     {
-                        return $"{(TotalDownloads / 1000):N0}k";
+                        return $"{(TotalDownloads / 1000):N0}{ThousandsChar}";
                     }
                 }
             }
 
             internal TemplateGroupTableRow TemplateGroupInfo { get; private set; }
 
             internal long TotalDownloads { get; private set; }
+
+            private class DownloadCountComparer : IComparer<string>
+            {
+                public int Compare(string? x, string? y)
+                {
+                    if (x == y || string.IsNullOrWhiteSpace(x) && string.IsNullOrWhiteSpace(y))
+                    {
+                        return 0;
+                    }
+                    if (string.IsNullOrWhiteSpace(x))
+                    {
+                        return -1;
+                    }
+                    if (string.IsNullOrWhiteSpace(y))
+                    {
+                        return 1;
+                    }
+                    int xInt = 0;
+                    int yInt = 0;
+
+                    if (x != MinimumDownloadCount)
+                    {
+                        int.TryParse(x.Substring(0, x.Length - 1), out xInt);
+                    }
+                    if (y != MinimumDownloadCount)
+                    {
+                        int.TryParse(y.Substring(0, y.Length - 1), out yInt);
```

Comment:
I wish we could store row data in a type safe way instead of parsing it back.

Reply:
We cannot - current formatter only works with strings. The alternative is to sort the collection before rendering.